src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -7910,4 +7933,243 @@ void wm_vuldel_truncate_revision(char * revision) {
+                    w_strdup(vulnerability->valuestring, (*alas_vuln)->fixed_packages[i]);
+                    (*alas_vuln)->fixed_packages[++i] = NULL;
+                }
+            }
+        }
+    }
+
+    return 0;
+}
+
+void wm_vuldet_fixed_packages_parser(vu_alas_vuln *vuln) {
+    vu_alas_vuln_pkg *new = NULL;
+    char **split_pkg = NULL;
+    int i = 0;
+
+    for (int j = 0; vuln->fixed_packages[j]; j++) {
+        os_calloc(1, sizeof(vu_alas_vuln_pkg), new);
+        char *pkg = vuln->fixed_packages[j];
+        char aux[OS_SIZE_256];
+        char *name = NULL;
+        char *version = NULL;
+        char *arch = NULL;
+        if (split_pkg = w_string_split(pkg, "-", 0), split_pkg) {
+            for (i = 0; split_pkg[i+2]; i++) {
+                wm_strcat(&name, split_pkg[i], '-');
+            }
+            snprintf(aux, OS_SIZE_256, "%s-%s", split_pkg[i], split_pkg[i+1]);
+            arch = strrchr(pkg, '.');
+            version = w_remove_substr(aux, arch);
+        }
+        w_strdup(name, new->pkg_name);
+        w_strdup(version, new->pkg_version);
+        w_strdup(arch+1, new->pkg_arch);
+
+        new->prev = vuln->vuln_pkg;
+        vuln->vuln_pkg = new;
+    }
+}
+
+int wm_vuldet_insert_ALAS(sqlite3 *db, vu_alas *alas, const char *target, int arch_id) {
+    sqlite3_stmt *stmt = NULL;
+    int result;
+    vu_alas_vuln *vul_it = alas->vuln;
+
+    for (; vul_it; vul_it = wm_vuldet_free_alas_vuln_node(vul_it)) {
+
+        if (vul_it->id) {
+
+            wm_vuldet_fixed_packages_parser(vul_it);
+
+            for (; vul_it->vuln_pkg; vul_it->vuln_pkg = wm_vuldet_free_alas_vuln_pkg_node(vul_it->vuln_pkg)) {
+
+                for (int i = 0; vul_it->vulnerabilities[i]; i++) {
+
+                    if (wm_vuldet_prepare(db, vu_queries[VU_INSERT_CVE], -1, &stmt, NULL) != SQLITE_OK) {
+                        return wm_vuldet_sql_error(db, stmt);
+                    }
+
+                    sqlite3_bind_text(stmt, 1, vul_it->vulnerabilities[i], -1, NULL);
+                    sqlite3_bind_text(stmt, 2, target, -1, NULL);
+                    sqlite3_bind_text(stmt, 3, NULL, -1, NULL);
+                    sqlite3_bind_text(stmt, 4, vul_it->vuln_pkg->pkg_name, -1, NULL);
+                    sqlite3_bind_text(stmt, 5, "less than", -1, NULL);
+                    sqlite3_bind_text(stmt, 6, vul_it->vuln_pkg->pkg_version, -1, NULL);
+                    sqlite3_bind_int(stmt, 7, 0);
+                    sqlite3_bind_int(stmt, 8, 0);
+                    sqlite3_bind_int(stmt, 9, ++arch_id);
+
+                    if (result = wm_vuldet_step(stmt), result != SQLITE_DONE && result != SQLITE_CONSTRAINT) {
+                        return wm_vuldet_sql_error(db, stmt);
+                    }
+
+                    wdb_finalize(stmt);
+
+                    // Insert architecture values
+                    if (arch_id && vul_it->vuln_pkg->pkg_arch) {
```

Comment:
This method, as well as for Redhat vulnerabilities, insert a lot of unnecessary entries in the `ARCHITECTURES` table, since it doesn't check if already exists any entry for the same target and architecture value.

We could improve this algorithm by:

1. Before assigning the `arch_id` to a particular CVE, query the architectures table for the ID of the target and architecture we are handling. If it returns a valid ID, we assign it to the CVE
2. If there no exists any ID for those values, we increase the `arch_id` and insert a new entry in the architectures table.